### PR TITLE
Add events searching methods

### PR DESCRIPTION
**Pull request by manbernotas**

### CODE REVIEW

File: EventPlaning.BL.Tests/EventManagerTests.cs [hunk 1]
```diff
@@ -162,91 +188,5 @@ public void GetEventIdFail()
...
-        //        Date = "2018-02-08",
-        //        Address = "Ateities g. 4F, Vilnius",
-        //        Duration = "5",
-        //        MinParticipants = 4,
-        //        MaxParticipants = 12,
-        //        Activities = new string[1] { "Monopoly" },
-        //    };
-
-        //    Assert.IsFalse(eventManager.CreateEvent(eventData));
-        //}
```
Comment (mantasmarcinkus) [inline EventPlaning.BL.Tests/EventManagerTests.cs:250]: not useful?

File: EventPlanning.DAL/Repository.cs [hunk 2]
```diff
@@ -53,6 +54,21 @@ public bool SaveActivityType(ActivityType activityType)
...
 
+        public IQueryable<Event> GetEvents(string pattern)
+        {
+            return context.Event
+                .Where(e => e.Address.Contains(pattern)
+                         || e.Title.Contains(pattern)
+                         || e.Description.Contains(pattern)
+                         || e.UserId.ToString().Contains(pattern)
+                         || e.Id.ToString().Contains(pattern));
+        }
```
Comment (mantasmarcinkus) [inline EventPlanning.DAL/Repository.cs]: Repository should not contain logic. Imagine we have a different database, which does not have this structure of model
Repository does not care how should it return data. It cares what data should it return: I return Events; I return Activities. But not which events (business logic should do that).

File: EventPlanning.DAL/Repository.cs [hunk 3]
```diff
@@ -53,6 +54,21 @@ public bool SaveActivityType(ActivityType activityType)
             return true;
         }
 
+        public IQueryable<Event> GetEvents(string pattern)
```
Comment (mantasmarcinkus) [inline EventPlanning.DAL/Repository.cs]: Is there any difference between `IQueryable` and `IEnumerable` in this case? https://stackoverflow.com/questions/2876616/returning-ienumerablet-vs-iqueryablet

File: EventPlanning.DAL/Repository.cs [hunk 4]
```diff
@@ -53,6 +54,21 @@ public bool SaveActivityType(ActivityType activityType)
...
+        {
+            return context.Event
+                .Where(e => e.Address.Contains(pattern)
+                         || e.Title.Contains(pattern)
```
Comment (mantasmarcinkus) [inline EventPlanning.DAL/Repository.cs]: What happens if `Title` is `null`?

File: EventPlanning.DAL/Repository.cs [hunk 5]
```diff
@@ -53,6 +54,21 @@ public bool SaveActivityType(ActivityType activityType)
...
+                         || e.Description.Contains(pattern)
+                         || e.UserId.ToString().Contains(pattern)
+                         || e.Id.ToString().Contains(pattern));
+        }
+
+        public IQueryable<Event> GetEvents(DateTime dateFrom, DateTime dateTo)
```
Comment (mantasmarcinkus) [inline EventPlanning.DAL/Repository.cs]: Same as above ☝️

File: EventPlanning.BL/EventManager.cs [hunk 6]
```diff
@@ -37,6 +37,27 @@ public List<Event> GetUserEvents(int userId)
...
+
+        /// <summary>
+        /// Returns events between date intervals
+        /// </summary>
+        /// <param name="dateFrom"></param>
+        /// <param name="dateTo"></param>
+        /// <returns></returns>
+        public List<Event> GetEvents(DateTime dateFrom, DateTime? dateTo)
+        {
+            return repository.GetEvents(dateFrom, dateTo ?? dateFrom).ToList();
```
Comment (mantasmarcinkus) [inline EventPlanning.BL/EventManager.cs]: What happens if `dateFrom` is sent as `2017-01-29 00:00:00`  which is a default `date` that is sent most of the time from Front end (no hours selected is defaulted as `00:00:00`

**Review by mantasmarcinkus — CHANGES_REQUESTED** [review]
Some changes 😂

File: EventPlaning.BL.Tests/EventManagerTests.cs (hunk 1 above)
Comment (manbernotas) [reply EventPlaning.BL.Tests/EventManagerTests.cs:250]: As we talked, any records creating methods can't be tested by using unit tests. Later will add integration tests.

Comment (mantasmarcinkus) [reply EventPlaning.BL.Tests/EventManagerTests.cs:250]: 👍

File: EventPlanning.service/Controllers/EventsController.cs [hunk 7]
```diff
@@ -31,9 +31,9 @@ public List<DAL.Event> GetEvents(int userId)
         {
             return eventManager.GetUserEvents(userId);
         }
-
-        // GET api/events/search
-        [HttpGet("search/{pattern}")]
+        
+        // GET api/events/search/Pan
+        [HttpGet("search/{pattern:minlength(3)}")]
```
Comment (mantasmarcinkus) [inline EventPlanning.service/Controllers/EventsController.cs]: 👍

File: EventPlanning.BL/EventManager.cs [hunk 8]
```diff
@@ -53,9 +59,11 @@ public List<Event> GetEvents(string pattern)
         /// <param name="dateFrom"></param>
         /// <param name="dateTo"></param>
         /// <returns></returns>
-        public List<Event> GetEvents(DateTime dateFrom, DateTime? dateTo)
+        public List<Event> GetEvents(DateTime dateFrom, DateTime dateTo)
         {
-            return repository.GetEvents(dateFrom, dateTo ?? dateFrom).ToList();
+            return repository.GetEvents()
+                .Where(e => e.DateFrom.Date >= dateFrom && e.DateTo <= dateTo.Date)
```
Comment (mantasmarcinkus) [inline EventPlanning.BL/EventManager.cs]: Missing tests for edge cases and dates

**Review by mantasmarcinkus — CHANGES_REQUESTED** [review]
Some tests missing
